File: 0071-simplify-path/0071-simplify-path.cs
public class Solution {
    public string SimplifyPath(string path) {
        // /a/./b/.../c/-> /c
        //Stack<string> stack for storig path.
        //string[] parts = path.Split('/'); ["","a",..]
        //foreach(string str in path){}
        Stack<string> stack = new Stack<string>();

        string[] parts = path.Split('/');   

        foreach(string part in parts){
          if( part == "." || part == ""){
            continue;
          }
          else if(part == ".."){
            if(stack.Count>0){
                 stack.Pop(); //remove prev parent

            }    
          }
          else{
            //valid, just add on path.
            stack.Push(part);
          }
        } 
        var result = "/" + string.Join("/", stack.Reverse());
        return result;
    }
}

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        parts = path.split('/')

        stack=[]

        for part in parts:
          if part == '' or part =='.':
            continue
          elif part == "..":
            if stack:
              stack.pop()
          else:
            stack.append(part)
        result = '/' + '/'.join(stack) # / + a/b..
        return result

          

